packages/core/components/NestedDropdown/NestedDropdown.tsx

```diff
@@ -102,12 +110,11 @@ const NestedDropdown: React.FC<NestedDropdownProps> = ({ data, tiers: [firstTier
 
   const chooseSelectedSecondTier = (tierOne: string, tierTwo: string) => {
     searchInput.current.focus()
-    const selectedItemValue = tierTwo
-    setUserSelectedTierTwoLabel(tierOne + tierTwo)
+    const selectedItemValue = `${tierOne} - ${tierTwo}`
     setUserSearchTerm('')
     setIsListOpened(false)
     setInputValue(selectedItemValue)
-    handleSelectedItems(tierOne, tierTwo)
+    handleSelectedItems(index, [tierOne, tierTwo])
```

Comment:
can you pass the index to the handleSelectedItems function before it's passed down to this component. That's they only reason you're passing index it seems